lib/charms/grafana_cloud_integrator/v0/cloud_config_provider.py

```diff
@@ -49,9 +49,6 @@ def _on_relation_changed(self, event):
         if not self._charm.unit.is_leader():
             return
         
-        if not self._charm.credentials_configured:
-            return
-
         for relation in self._charm.model.relations[self._relation_name]:
             databag = relation.data[self._charm.app]
             for k, v in (
```

Comment:
This for-loop is longer than direct assignment:
```python
databag["username"] = ...
```

Reply:
Done!